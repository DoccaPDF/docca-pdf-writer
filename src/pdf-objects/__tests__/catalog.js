import Catalog from '../catalog';
import Names from '../names';

describe('pdf-objects', () => {
  describe('catalog', () => {
    it('creates a new object', () => {
      const content = Catalog({
        id: 2,
        Pages: { id: 3 }
      });
      expect(content).toEqual(expect.objectContaining({
        Type: 'Catalog',
        Pages: `3 0 R`
      }));
    });

    it('flattens keys values', () => {
      const content = Catalog({
        id: 2,
        Pages: { id: 3 },
        Names: Names()
      });
      content.addNameDest({ name: 'myDest', page: { id: 1 }, left: 10, top: 10, zoom: 2 });
      expect(content.flattenKeys()).toMatchSnapshot();
    });

    it('returns a PDF object', () => {
      const content = Catalog({
        id: 2,
        Pages: { id: 3 },
        Names: Names()
      });
      content.addNameDest({ name: 'myDest', page: { id: 1 }, left: 10, top: 10, zoom: 2 });
      expect(content.toPDF()).toMatchSnapshot();
    });
  });
});
