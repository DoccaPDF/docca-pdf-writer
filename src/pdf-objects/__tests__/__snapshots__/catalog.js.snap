// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pdf-objects catalog flattens keys values 1`] = `
Object {
  "Names": "<<
/Dests << /Names [(myDest) [1 0 R /XYZ 10 10 2]] >>
>>
",
  "Pages": "3 0 R",
  "Type": "/Catalog",
}
`;

exports[`pdf-objects catalog returns a PDF object 1`] = `
"2 0 obj
<<
/Type /Catalog
/Pages 3 0 R
/Names <<
/Dests << /Names [(myDest) [1 0 R /XYZ 10 10 2]] >>
>>

>>
endobj
"
`;
